### PR TITLE
ipq40xx: add support for Netgear SRR60/SRS60 and RBR50/RBS50

### PR DESCRIPTION
**Pull request by DavideFioravanti**
These patches add support for Netgear SRR60/SRS60 and RBR50/RBS50 and mmc based devices in general.

The patch [base-files: add minimal mmc support](https://github.com/openwrt/openwrt/commit/878f465cb4c228bf6a8775150009a6c4a3b22061) can be used to standardize the upgrade procedure for mmc devices like the Zyxel NBG6817

~~For what concerns the [ipq40xx: add support for Netgear SRR60/SRS60 and RBR50/RBS50](https://github.com/openwrt/openwrt/commit/e83244a78f63cf85c63ed900ae40fcbefe260881) there is currently a "bug" in the IPQ40XX switch driver which prevents the port 1 to correctly detect the cable connection [(Read more)](https://forum.openwrt.org/t/netgear-orbi-pro-support-booting-from-mmc/14208/209).
I don't know if a fix for this problem will be ready before the introduction of the DSA driver for ipq40xx, so in the meantime I set `qcom,poll_required = <0>;` for the WAN port. The only downside of this is that the WAN port always results as connected even if not.~~
~~When this [PR](https://github.com/openwrt/openwrt/pull/3498) will be accepted, the wan port problem will be solved.~~
Merged. WAN port is working now.

### CODE REVIEW

**Comment by DavideFioravanti** [comment]
@robimarko As I wrote in the forum, I think that you should be added as co-developer of this patch giving the fact that lot of my work is based on yours, but you didn't respond. I would add you, but I don't know if it's the right thing to do without your consensus... :thinking:

**Comment by robimarko** [comment]
Hi, I was not really following the thread after I had to return the SRK60 pair I had for testing.
Feel free adding my SoB:
Signed-off-by: Robert Marko <robimarko@gmail.com>

About the switch driver, yeah that one is a mess of old Qualcomm GPL sources that were updated to work with new kernels.
There is some work to switch to QCA8k based DSA one but it will take a lot of work and time.

I see that you figured out the cmdline partition trick, I tried that also but forgot that it needs to be enabled first.
I am glad to see this thing alive and kicking.

File: package/base-files/files/lib/upgrade/mmc.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#!/bin/sh
```
Comment (adschm) [inline package/base-files/files/lib/upgrade/mmc.sh:1]: That's a library file, remove the shebang and the execute bit (if there is one).

File: package/base-files/files/lib/upgrade/mmc.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+. /lib/functions.sh
+
+mmc_do_upgrade() {
+
+    command -v losetup >/dev/null || {
```
Comment (adschm) [inline package/base-files/files/lib/upgrade/mmc.sh:7]: broken indent here. I would also remove the empty line before.

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+			gpios = <&tlmm 49 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		power-green {
+			label = "orbi:power:green";
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:103]: Typically, the color is in the middle, so I would except
orbi:green:power
here.

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+		};
+
+		power-red {
+			label = "orbi:power:red";
+			gpios = <&tlmm 64 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
+			panic-indicator;
+		};
+
+		green: green {
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:115]: These names/labels are inconsistent. I'd use something like

led_status_green: status_green {

or

led_status_green: status-green {

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+	status = "okay";
+	
+	tlc59208f@27 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "ti,tlc59108";
+		reg = <0x27>;
+		
+		led0@0 {
+			label = "orbi:led0:rgb";
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:225]: See above for color:function order. I have never dealt with rgb leds though, so I don't know what's the best practice there.

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbr50.dts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+/ {
+    model = "Netgear RBR50";
+	compatible = "netgear,rbr50";
+
+	chosen {
+		bootargs = "root=/dev/mmcblk0p20 blkdevparts=mmcblk0:512K@17K(0:SBL1)ro,512K(0:BOOTCONFIG)ro,512K(0:QSEE)ro,512K(0:QSEE_ALT)ro,256K(0:CDT)ro,256K(0:CDT_ALT)ro,256K(0:DDRPARAMS)ro,256K(0:APPSBLENV)ro,1M(0:APPSBL)ro,1M(0:APPSBL_ALT)ro,256K(0:ART)ro,256K(ARTMTD)ro,2M(language)ro,256K(config)ro,256K(pot)ro,256K(traffic_meter)ro,256K(pot_bak)ro,256K(traffic_meter.bak)ro,3840K(kernel),31488K(rootfs),35328K@9233K(firmware),256K(mtdoops)ro,1457651200(reserved)ro,-(unallocated) rootfstype=squashfs,ext4 rootwait";
+	};
+
+	soc {
+
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbr50.dts:14]: remove empty line.

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbr50.dts [hunk 7]
```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qcom-ipq4019-orbi.dtsi"
+
+/ {
+    model = "Netgear RBR50";
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbr50.dts:6]: indent.

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts [hunk 8]
```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qcom-ipq4019-orbi.dtsi"
+
+/ {
+    model = "Netgear RBS50";
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts:6]: indent.

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+	compatible = "netgear,rbs50";
+
+	chosen {
+		bootargs = "root=/dev/mmcblk0p20 blkdevparts=mmcblk0:512K@17K(0:SBL1)ro,512K(0:BOOTCONFIG)ro,512K(0:QSEE)ro,512K(0:QSEE_ALT)ro,256K(0:CDT)ro,256K(0:CDT_ALT)ro,256K(0:DDRPARAMS)ro,256K(0:APPSBLENV)ro,1M(0:APPSBL)ro,1M(0:APPSBL_ALT)ro,256K(0:ART)ro,256K(ARTMTD)ro,2M(language)ro,256K(config)ro,256K(pot)ro,256K(traffic_meter)ro,256K(pot_bak)ro,256K(traffic_meter.bak)ro,3840K(kernel),31488K(rootfs),35328K@9233K(firmware),256K(mtdoops)ro,1457651200(reserved)ro,-(unallocated) rootfstype=squashfs,ext4 rootwait";
+	};
+
+	soc {
+
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts:14]: remove empty line

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-srr60.dts [hunk 10]
```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qcom-ipq4019-orbi.dtsi"
+
+/ {
+    model = "Netgear SRR60";
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-srr60.dts:6]: indent; and for the next file again.

File: target/linux/ipq40xx/image/Makefile [hunk 11]
```diff
@@ -617,6 +617,63 @@ define Device/netgear_ex6150v2
 endef
 TARGET_DEVICES += netgear_ex6150v2
 
+define Device/netgear_orbi
+	$(call Device/DniImage)
+	SOC := qcom-ipq4019
+	DEVICE_VENDOR := Netgear
+	BLOCKSIZE := 128k
+	IMAGE/factory.img := append-kernel | pad-offset $$$$(BLOCKSIZE) 64 | append-uImage-fakehdr filesystem | pad-to $$$$(KERNEL_SIZE) | append-rootfs | pad-rootfs | netgear-dni
```
Comment (adschm) [inline target/linux/ipq40xx/image/Makefile:625]: Please wrap lines.

File: target/linux/ipq40xx/image/Makefile [hunk 12]
```diff
@@ -617,6 +617,63 @@ define Device/netgear_ex6150v2
...
+	DEVICE_PACKAGES := ath10k-firmware-qca9984-ct e2fsprogs kmod-fs-ext4 losetup
+endef
+
+define Device/netgear_rbx50
+	$(call Device/netgear_orbi)
+	NETGEAR_HW_ID := 29765352+0+4000+512+2x2+2x2+4x4
+	KERNEL_SIZE := 3932160
+	ROOTFS_SIZE := 32243712
+	IMAGE_SIZE := 36175872
+	DEVICE_VENDOR := Netgear
```
Comment (adschm) [inline target/linux/ipq40xx/image/Makefile:637]: You already have the vendor in netgear_orbi

File: target/linux/ipq40xx/image/Makefile [hunk 13]
```diff
@@ -617,6 +617,63 @@ define Device/netgear_ex6150v2
...
+endef
+TARGET_DEVICES += netgear_rbs50
+
+define Device/netgear_srx60
+	$(call Device/netgear_orbi)
+	NETGEAR_HW_ID := 29765352+0+4096+512+2x2+2x2+4x4
+	KERNEL_SIZE := 3932160
+	ROOTFS_SIZE := 32243712
+	IMAGE_SIZE := 36175872
+	DEVICE_VENDOR := Netgear
```
Comment (adschm) [inline target/linux/ipq40xx/image/Makefile:660]: already in netgear_orbi

File: target/linux/ipq40xx/image/Makefile [hunk 14]
```diff
@@ -617,6 +617,63 @@ define Device/netgear_ex6150v2
...
+endef
+
+define Device/netgear_rbr50
+	$(call Device/netgear_rbx50)
+	NETGEAR_BOARD_ID := RBR50
+	DEVICE_MODEL := RBR50
```
Comment (adschm) [inline target/linux/ipq40xx/image/Makefile:643]: I'd put DEVICE_MODEL before NETGEAR_BOARD_ID here and for the others below.

File: package/base-files/files/lib/upgrade/mmc.sh (hunk 1 above)
Comment (DavideFioravanti) [reply package/base-files/files/lib/upgrade/mmc.sh]: Removed, thanks!

File: package/base-files/files/lib/upgrade/mmc.sh (hunk 2 above)
Comment (DavideFioravanti) [reply package/base-files/files/lib/upgrade/mmc.sh]: Fixed, thanks!

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi (hunk 3 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi]: Fixed, thanks!

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi (hunk 4 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi]: Fixed, thanks!

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi (hunk 5 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi]: The color of these leds is set by the GPIOS defined before... so these "rgb" leds definitions are used just to turn on or off the leds

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbr50.dts (hunk 6 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbr50.dts]: Fixed, thanks!

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbr50.dts (hunk 7 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbr50.dts]: Fixed, thanks!

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts (hunk 8 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts]: Fixed, thanks!

Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts]: Fixed, thanks!

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts (hunk 9 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts]: Fixed, thanks!

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-srr60.dts (hunk 10 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-srr60.dts]: Fixed, thanks!

File: target/linux/ipq40xx/image/Makefile (hunk 11 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/image/Makefile]: Fixed, thanks!

File: target/linux/ipq40xx/image/Makefile (hunk 12 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/image/Makefile]: Fixed, thanks!

File: target/linux/ipq40xx/image/Makefile (hunk 13 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/image/Makefile]: Fixed, thanks!

File: target/linux/ipq40xx/image/Makefile (hunk 14 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/image/Makefile:646]: Fixed, thanks!

**Comment by DavideFioravanti** [comment]
@adschm I am really sorry for the errors in the commit. :disappointed: I forgot to check it with checkpatch.pl...

**Comment by DavideFioravanti** [comment]
Rebased to current master and updated the wan port configuration to make it working correctly when https://github.com/openwrt/openwrt/pull/3498 will be accepted.
I'll remove the draft status of this PR when the aforementioned PR will be upstreamed

**Comment by robimarko** [comment]
@DavideFioravanti I hope it will get merged soon, so I can stop rebasing multiple PR-s that depend on it now.
I also have some EDMA fixes as well as fiber support queued as soon as it's merged.

**Comment by DavideFioravanti** [comment]
I hope so too... And above all, I hope that will be merged before the 20.x branch :crossed_fingers:

**Comment by robimarko** [comment]
@DavideFioravanti Can you add your Tested-by tag to my PHY PR?
https://github.com/openwrt/openwrt/pull/3498

That would highly increase the chances of it getting merged.

**Comment by DavideFioravanti** [comment]
Rebased and tested again cherry-picking these commits: https://github.com/openwrt/openwrt/pull/3498. It's working correctly.

**Comment by DavideFioravanti** [comment]
Rebased and tested after the merge of this PR https://github.com/openwrt/openwrt/pull/3498

**Comment by theoribeiro** [comment]
@DavideFioravanti and @robimarko 
Commented on the forum post but not sure if you're following the topic. 
I'm testing this PR on my RBS50 (v2 if I'm not mistaken) and getting the following output on UART:
```
Format: Log Type - Time(microsec) - Message - Optional Info
Log Type: B - Since Boot(Power On Reset),  D - Delta,  S - Statistic
S - QC_IMAGE_VERSION_STRING=BOOT.BF.3.1.1-00123
S - IMAGE_VARIANT_STRING=DAACANAZA
S - OEM_IMAGE_VERSION_STRING=CRM
S - Boot Config, 0x00000025
S - Reset status Config, 0x00000000
S - Core 0 Frequency, 0 MHz
B -       261 - PBL, Start
B -      1338 - bootable_media_detect_entry, Start
B -      2608 - bootable_media_detect_success, Start
B -      2622 - elf_loader_entry, Start
B -      4028 - auth_hash_seg_entry, Start
B -      6177 - auth_hash_seg_exit, Start
B -     74171 - elf_segs_hash_verify_entry, Start
B -    197602 - PBL, End
B -    197626 - SBL1, Start
B -    291052 - pm_device_init, Start
D -         7 - pm_device_init, Delta
B -    292552 - boot_flash_init, Start
D -     95234 - boot_flash_init, Delta
B -    392087 - boot_config_data_table_init, Start
D -     13941 - boot_config_data_table_init, Delta - (419 Bytes)
B -    408780 - clock_init, Start
D -      7562 - clock_init, Delta
B -    419783 - CDT version:2,Platform ID:8,Major ID:1,Minor ID:0,Subtype:1
B -    423195 - sbl1_ddr_set_params, Start
B -    428292 - cpr_init, Start
D -         2 - cpr_init, Delta
B -    432674 - Pre_DDR_clock_init, Start
D -         4 - Pre_DDR_clock_init, Delta
D -     13173 - sbl1_ddr_set_params, Delta
B -    445986 - pm_driver_init, Start
D -         2 - pm_driver_init, Delta
B -    516547 - sbl1_wait_for_ddr_training, Start
D -        29 - sbl1_wait_for_ddr_training, Delta
B -    531946 - Image Load, Start
D -    145927 - QSEE Image Loaded, Delta - (274048 Bytes)
B -    678300 - Image Load, Start
D -      2114 - SEC Image Loaded, Delta - (2048 Bytes)
B -    688506 - Image Load, Start
D -    179397 - APPSBL Image Loaded, Delta - (424967 Bytes)
B -    868297 - QSEE Execution, Start
D -        56 - QSEE Execution, Delta
B -    874471 - SBL1, End
D -    678955 - SBL1, Delta
S - Flash Throughput, 2040 KB/s  (701834 Bytes,  343936 us)
S - DDR Frequency, 672 MHz


U-Boot 2012.07-25708-gfbe2156 [Chaos Calmer 15.05.1,r35193] (Oct 25 2018 - 15:15:28)

U-boot dni1 V0.5 for DNI HW ID: 29765913; NOR flash 0MB; NAND flash 128MB; RAM 256MB
smem ram ptable found: ver: 1 len: 3
DRAM:  256 MiB
machid : 0x8010001
NAND:  ID = 9500f1ef
Vendor = ef
Device = f1
ONFI device found
SF NAND unsupported id:ff:ff:ff:ffSF: Unsupported manufacturer ff
ipq_spi: SPI Flash not found (bus/cs/speed/mode) = (0/0/48000000/0)
128 MiB
MMC:   qca_mmc: 0
*** Warning - bad CRC, using default environment

PCI0 Link Intialized
In:    serial
Out:   serial
Err:   serial
Valid chip addresses: 27 48 4A
machid: 8010001
flash_type: 2
Net:    131072 bytes read: OK
MAC0 addr:bc:a5:11:b2:91:c3
PHY ID1: 0x4d
PHY ID2: 0xd0b1
ipq40xx_ess_sw_init done
eth0
Hit any key to stop autoboot:  0
eth0 PHY0 Down Speed :10 Half duplex
eth0 PHY1 Down Speed :10 Half duplex
eth0 PHY2 Down Speed :10 Half duplex
eth0 PHY3 Down Speed :10 Half duplex
eth0 PHY4 Down Speed :10 Half duplex

 nmrp server is stopped or failed !
Loading DNI firmware for checking...
Loading firmware 1 ...

NAND read: device 0 offset 0x3280000, size 0x20000
 131072 bytes read: OK

NAND read: device 0 offset 0x3280000, size 0x2a0000
 2752512 bytes read: OK
dniimg_len is 2752512 (aligned to 2752512)

NAND read: device 0 offset 0x3280000, size 0x2a0000
 2752512 bytes read: OK
if iminfo 0x84000000; then echo kernel checksum OK !;if iminfo 0x8429FFC0; then echo rootfs checksum OK !;boot_partition_set 1;true;else echo rootfs checksum error !!!;fw_recovery;false;fi;else echo kernel checksum error !!!;fw_recovery;false;fi;
## Checking Image at 84000000 ...
   FIT image found
   FIT description: ARM OpenWrt FIT (Flattened Image Tree)
    Image 0 (kernel@1)
     Description:  ARM OpenWrt Linux-5.4.86
     Type:         Kernel Image
     Compression:  uncompressed
     Data Start:   0x840000e4
     Data Size:    2702640 Bytes = 2.6 MiB
     Architecture: ARM
     OS:           Linux
     Load Address: 0x80208000
     Entry Point:  0x80208000
     Hash algo:    crc32
     Hash value:   7d831d33
     Hash algo:    sha1
     Hash value:   3ebef3573af8efd71157277ceebe052796af6393
    Image 1 (fdt@1)
     Description:  ARM OpenWrt netgear_rbs50 device tree blob
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x84293f4c
     Data Size:    17925 Bytes = 17.5 KiB
     Architecture: ARM
     Hash algo:    crc32
     Hash value:   de7c73b1
     Hash algo:    sha1
     Hash value:   fdd0712c2ba0efa2ef02faa2d841a16611f668bc
    Default Configuration: 'config@1'
    Configuration 0 (config@1)
     Description:  OpenWrt
     Kernel:       kernel@1
     FDT:          fdt@1
## Checking hash(es) for FIT Image at 84000000 ...
   Hash(es) for Image 0 (kernel@1): crc32+ sha1+
   Hash(es) for Image 1 (fdt@1): crc32+ sha1+
kernel checksum OK !

## Checking Image at 8429ffc0 ...
   Legacy image found
   Image Name:   OpenWrt fake filesystem
   Image Type:   ARM Linux Filesystem Image (uncompressed)
   Data Size:    0 Bytes = 0 Bytes
   Load Address: 00000000
   Entry Point:  00000000
   Verifying Checksum ... OK
rootfs checksum OK !
Burn BOOT PARTITION (= 1) into boarddata2 block
 131072 bytes read: OK
Erasing: off d80000, size 20000
Erasing at 0xd80000 -- 100% complete. Cleanmarker written at 0xd80000.
OK
Writing: from RAM addr 871cf9c8, to NAND off d80000, size 20000
 131072 bytes written: OK
Alive-timer 11
Done.
## Booting kernel from FIT Image at 84000000 ...
ERROR: can't get kernel image!
```

I can't help much with understanding what is going on but I'm glad to try things on this device and provide outputs if it can help. Let me know if there's anything you need me to try! 
Thanks for your efforts so far!

**Comment by DavideFioravanti** [comment]
> Commented on the forum post but not sure if you're following the topic.
I read your comment but I didn't answer because I am very busy at the moment. Sorry.

> I'm testing this PR on my RBS50 (v2 if I'm not mistaken)

I wasn't aware of the v2 model. In the following days I will update this commit to specify the v1 version in the name.

However reading your bootlog it seems to not be compatible with this commit because your router uses NAND flash instead of the MMC of the v1. Unfortunately I can't do anything about it.

File: target/linux/ipq40xx/base-files/etc/board.d/02_network [hunk 15]
```diff
@@ -104,6 +104,14 @@ ipq40xx_setup_interfaces()
 			"0u@eth0" "3:lan:2" "4:lan:1"
 		ucidef_set_interface_wan "eth1"
 		;;
+	netgear,rbr50|\
+	netgear,rbs50|\
+	netgear,srr60|\
+	netgear,srs60)
+		ucidef_set_interfaces_lan_wan "eth0" "eth1"
```
Comment (adschm) [inline target/linux/ipq40xx/base-files/etc/board.d/02_network:111]: Just using ucidef_set_interface_wan should be enough, the lan should be picked up from the switch command.

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi [hunk 16]
```diff
@@ -0,0 +1,326 @@
...
+			gpios = <&tlmm 49 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		power_green {
+			label = "orbi:green:power";
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:103]: Please remove the "orbi:" prefix for all leds, we just use color:function now ...

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi [hunk 17]
```diff
@@ -0,0 +1,326 @@
...
+			gpios = <&tlmm 63 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+
+		power_red {
+			label = "orbi:red:power";
+			gpios = <&tlmm 64 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:111]: default-state=off is default and can be dropped.

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi [hunk 18]
```diff
@@ -0,0 +1,326 @@
...
+	status = "okay";
+
+	tlc59208f@27 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "ti,tlc59108";
+		reg = <0x27>;
+
+		led0@0 {
+			label = "orbi:rgb:led0";
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:225]: same here, drop orbi: prefix for all leds.

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi [hunk 19]
```diff
@@ -0,0 +1,326 @@
...
+
+	status = "okay";
+
+	tlc59208f@27 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "ti,tlc59108";
+		reg = <0x27>;
+
+		led0@0 {
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:224]: Just led@0, led@1 etc.?

File: target/linux/ipq40xx/image/Makefile [hunk 20]
```diff
@@ -650,6 +650,62 @@ define Device/netgear_ex6150v2
 endef
 TARGET_DEVICES += netgear_ex6150v2
 
+define Device/netgear_orbi
+	$(call Device/DniImage)
+	SOC := qcom-ipq4019
+	DEVICE_VENDOR := Netgear
```
Comment (adschm) [inline target/linux/ipq40xx/image/Makefile:656]: Please use all-caps "NETGEAR"

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts [hunk 21]
```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qcom-ipq4019-orbi.dtsi"
+
+/ {
+	model = "Netgear RBS50";
```
Comment (adschm) [inline target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts:6]: Please use all-caps NETGEAR for model in all DTS files for consistency with existing devices.

Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts]: Fixed, thanks!

File: target/linux/ipq40xx/image/Makefile (hunk 20 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/image/Makefile]: Fixed, thanks!

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi (hunk 19 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:223]: Those leds create a ring on the top of the router and they are not labeled at all. They change color depending on which GPIO in the previous led section is HIGH.
image
I don't know how to name them.

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi (hunk 18 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi]: Fixed, thanks!

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi (hunk 17 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi]: Fixed, thanks!

File: target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi (hunk 16 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi]: Fixed, thanks!

File: target/linux/ipq40xx/base-files/etc/board.d/02_network (hunk 15 above)
Comment (DavideFioravanti) [reply target/linux/ipq40xx/base-files/etc/board.d/02_network]: I tested with your suggested changes and it's working correclty. Thanks

**Comment by DavideFioravanti** [comment]
Fixed, rebased and tested again.
I also added the V1 version for RBR50 and RBS50 to avoid any confusion:
image
According to the Netgear website currently the SRS60 and SRR60 don't have a V2 version.

**Comment by Bixilon** [comment]
@DavideFioravanti Thanks, but before anybody reviews this, you need to solve the merge conflicts

**Comment by DavideFioravanti** [comment]
Sorry for the delay. The conflicts were caused by new devices added.
I rebased, fixed the conflicts and tested again.

**Comment by Bixilon** [comment]
Any updates?

**Comment by Flole998** [comment]
Can this get reviewed and merged please?

**Comment by Flole998** [comment]
@adschm Could you please review this one again? It's ready to be merged IMO.

**Comment by chunkeey** [comment]
Ok, let's give this a try. 

I changed various things (and added 5.10). It would be great if the cmdline partition string could be shorter. Merged!

**Comment by ArtemiisFowl** [comment]
I have a netgear ex8000, it should be hardware identical to obri rbr50/rbs50 devices. What's my todo to make this device compatible with openwrt?

**Comment by Flole998** [comment]
Try to run the initramfs and see what happens.

**Comment by J0hnMatrix** [comment]
Hello,
I have RBK50v2 (without USB port), is there something I can provide for it's integration in this repo?

**Comment by Flole998** [comment]
Come up with a PR for it and it'll be merged eventually at some point....

**Pull request closed without merging**